src/tools/rest-backend/service_storage.cpp

```diff
@@ -0,0 +1,606 @@
+#include <regex>
+
+#include "service.hpp"
+
+namespace kdbrest
+{
+
+namespace service
+{
+
+/**
+         * @brief Default constructor that pre-fetches the cache.
+         */
+StorageEngine::StorageEngine ()
+{
+	// pre fetch the entry cache
+	this->loadAllEntries ();
+	// pre-fetch the user cache
+	this->loadAllUsers ();
+}
+
+/**
+         * @brief Can be used to create an entry in the database. Will add the
+         * entry and all subkeys to the database (configuration).
+         * 
+         * @param entry A custom Entry object holding information to store.
+         * 
+         * @return true if the entry was stored, false if something went wrong
+         * 
+         * @throws kdbrest::exception::EntryAlreadyExistsException in case an
+         * entry with the given name does already exist in the key database.
+         */
+bool StorageEngine::createEntry (model::Entry & entry)
+{
+	using namespace kdb;
+
+	// register exclusive access
+	boost::unique_lock<boost::shared_mutex> lock (_mutex_entry_cache);
+
+	std::vector<model::Entry> & entries = this->_entryCache;
+	for (auto & elem : entries)
+	{
+		if (elem.getPublicName ().compare (entry.getPublicName ()) == 0)
+		{
+			throw exception::EntryAlreadyExistsException ();
+		}
+	}
+
+	KDB kdb;
```

Comment:
No, but it doesn't work otherwise without the cachefilter yet. This is one of the things that can be changed then..